src/cljs/main/broadfcui/page/groups/groups_management.cljs

```diff
@@ -0,0 +1,129 @@
+    [broadfcui.utils :as utils]
+    ))
+
+
+(react/defc GroupTable
+  {:reload
+   (fn [{:keys [this]}]
+     (react/call :load-data this))
+   :render
+   (fn [{:keys [state this]}]
+     (cond
+       (:error-message @state) (style/create-server-error-message (:error-message @state))
+       (nil? (:groups @state)) [comps/Spinner {:text "Loading groups..."}]
+       :else
+       [Table
+        {:data (:groups @state)
+         :body {:behavior {:reorderable-columns? false}
+                :style table-style/table-light
+                :columns
+                [{:header "Group Name" :initial-width 500 :sort-initial :asc
+                  :sort-by :text
+                  :as-text :group-name
+                  :render
+                  (fn [{:keys [groupName accessLevels]}]
+                    (if
+                     (contains? (set accessLevels) "Owner")
+                      (style/create-link {:text groupName
+                                          :href (nav/get-link :group groupName)})
+                      groupName))}
+                 {:header "Role" :initial-width 100 :resizable? false
+                  :column-data #(clojure.string/join ", " (:accessLevels %))}
+                 {:id "delete group" :initial-width :auto
+                  :filterable? false :sortable? false
+                  :as-text
+                  (fn [{:keys [groupName accessLevels]}]
+                    (when (contains? (set accessLevels) "Owner")
+                      (str "Delete group " groupName)))
+                  :render
+                  (fn [{:keys [groupName accessLevels]}]
+                    (when (contains? (set accessLevels) "Owner")
+                      (style/create-link
+                       {:text (icons/icon {} :delete)
+                        :style {:float "right"}
+                        :onClick (fn []
+                                   (swap! state assoc :deleting? true)
+                                   (endpoints/call-ajax-orch
+                                    {:endpoint (endpoints/delete-group groupName)
+                                     :on-done (fn [{:keys [success? get-parsed-response]}]
+                                                (swap! state dissoc :deleting?)
+                                                (if success?
+                                                  (this :load-data)
+                                                  (swap! state assoc :error-mesage (get-parsed-response false))))}))})))}]}
+         :toolbar
+         {:items
+          [flex/spring
+           [comps/Button
+            {:text "Create New Group"
+             :onClick
+             (fn []
+               (modal/push-modal
+                [CreateGroupDialog
+                 {:on-success #(react/call :reload this)}]))}]]}}]))
+   :component-did-mount
+   (fn [{:keys [this]}]
+     (react/call :load-data this))
+   :load-data
+   (fn [{:keys [state]}]
+     (endpoints/get-groups
```

Comment:
This is the only usage of this function. What is the value of moving into a separate file (besides following the existing convention)?